### PR TITLE
For #13052 - Creates new dialog fragment for search and puts it behind a feature flag

### PR DESCRIPTION
**Pull request by boek**

### CODE REVIEW

File: app/src/main/res/layout/fragment_search_dialog.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    android:layout_height="match_parent"
+    xmlns:app="http://schemas.android.com/apk/res-auto">
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        android:text="Hello World"
```
Comment (ekager) [inline app/src/main/res/layout/fragment_search_dialog.xml:16]: beautiful new search screen 🤩

File: app/src/main/java/org/mozilla/fenix/FeatureFlags.kt [hunk 2]
```diff
@@ -38,4 +38,9 @@ object FeatureFlags {
      * Enables viewing tab history
      */
     val tabHistory = Config.channel.isNightlyOrDebug
+
+    /**
+     * Enables viewing tab history
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/FeatureFlags.kt:43]: Update description

File: app/src/main/res/values/static_strings.xml [hunk 3]
```diff
@@ -32,7 +32,7 @@
 
     <!-- Label for the secret settings preference -->
     <string name="preferences_debug_settings">Secret Settings</string>
-    <string name="preferences_debug_settings_enable_tab_tray">Use New Tab Tray</string>
+    <string name="preferences_debug_settings_use_new_search_experience">Use New Search Experience</string>
```
Comment (eliserichards) [inline app/src/main/res/values/static_strings.xml:35]: Add a description comment here

File: app/src/main/res/layout/fragment_search_dialog.xml (hunk 1 above)
Comment (boek) [reply app/src/main/res/layout/fragment_search_dialog.xml:16]: ty!!!

**Pull request merged**